spec/enumerable_spec.rb

```diff
@@ -33,28 +33,97 @@
     it 'iterates through each value and index in an array' do
-      i=0
-      array.my_each_with_index do |value, index|
+      i = 0
+      array.my_each_with_index do |_value, index|
         expect(index).to eq(i)
-      i+=1
-      end  
-    end  
+        i += 1
+      end
+    end
 
     it 'Returns to Enumerator if no block is given' do
       expect(array.my_each_with_index.is_a?(Enumerator)).to be true
     end
   end
+
   describe '#my_select' do
     it 'selects odd numbers from the array' do
-      expect(array.my_select(&:odd?)).to eq([5,11, 31])
+      expect(array.my_select(&:odd?)).to eq([5, 11, 31])
     end
 
     it 'Returns to Enumerator if no block is given' do
       expect(array.my_select.is_a?(Enumerator)).to be true
     end
 
     it 'select a key value pair from the hash' do
-      expect(hash.select { |key, value| value == 'bark' }).to eq({'dog' => 'bark'})
+      expect(hash.my_select { |_key, value| value == 'bark' }).to eq([%w[dog bark]])
+    end
+
+    it 'Returns to Enumerator if no block is given' do
+      expect(hash.my_select.is_a?(Enumerator)).to be true
+    end
+  end
+
+  describe '#my_all' do
+    it 'returns false if all elements are not odd' do
+      expect(array.my_all?(&:odd?)).to eq(false)
+    end
+
+    it 'returns true if no block given and no element is false or nil' do
+      expect(array.my_all?).to eq(true)
+    end
+  end
+
+  describe '#my_any' do
+    it 'returns true if any value returned by block is true' do
+      expect(array.my_any? { |v| v == 5 }).to eq(true)
+    end
+
+    it 'check for false element' do
+      expect(array.my_any? { |v| v == false }).to eq(false)
+    end
+  end
+
+  describe '#my_none' do
+    it 'returns true if all elements are false' do
+      expect(array.my_none? { |v| v.instance_of?(String) }).to eq(true)
+    end
+  end
+
+  describe '#my_count' do
+    it 'counts the number of elements in an array' do
+      expect(array.my_count).to eq(9)
+    end
+
+    it 'counts the number of even numbers in an array' do
+      expect(array.my_count(&:even?)).to eq(6)
+    end
+  end
+
+  describe '#my_map' do
+    it 'Returns the double of each number in an array' do
+      expect(array.my_map { |a| a * 2 }).to eql([4, 10, 12, 22, 140, 8, 44, 136, 62])
+    end
+
+    it 'it returns to an enumerator if no block is given' do
+      expect(array.my_map.is_a?(Enumerator)).to be true
+    end
+  end
+
+  describe '#my_inject' do
+    it 'returns the sum of the numbers in an array' do
+      expect(array.my_inject(0) { |sum, n| sum + n }).to eq(219)
+    end
+
+    it 'returns the product of the numbers in an array' do
+      expect(array.my_inject(1) { |product, n| product * n }).to eq(8_570_284_800)
+    end
+
+    it 'returns the sum of the elements in an array' do
+      expect(array.my_inject(:+)).to eq(219)
+    end
+
+    it 'returns the proper sum that includes the argument' do
+      expect { array.my_inject(3, :+) }.to raise_exception
     end
```

Comment:
- Despite the test is passing, it is expecting the wrong behavior. That call for my inject method should return an expected result and not raising an error.